emperor/support_files/js/sceneplotview3d.js

```diff
@@ -100,6 +102,90 @@ define([
     this.dimensionRanges = {'max': [], 'min': []};
     this.drawAxesWithColor(0xFFFFFF);
     this.drawAxesLabelsWithColor(0xFFFFFF);
+
+    var raycaster = new THREE.Raycaster();
+    var mouse = new THREE.Vector2();
+
+    //initialize subscribers for event callbacks
+    this.EVENTS = ['click', 'dblclick'];
+    var subscribers = {};
+    for (var i = 0; i < this.EVENTS.length; i++) {
+      subscribers[this.EVENTS[i]] = [];
+    }
+
+    //create helper function for callbacks
+    var eventCallback = function(eventType, event) {
+      event.preventDefault();
+      //dont do anything if no subscribers
+      if (subscribers[eventType].length === 0) {
+        return;
+      }
+
+      var element = scope.renderer.domElement;
+      mouse.x = ((event.clientX - element.offsetLeft) / element.width) * 2 - 1;
+      mouse.y = -((event.clientY - element.offsetTop) / element.height) * 2 + 1;
+
+      raycaster.setFromCamera(mouse, scope.camera);
```

Comment:
@wasade, I believe I understand what you are referring to regarding the decomposition of the event callbacks, but at this point, we are getting away by calling the same handler irregardless of the originating event i.e. `click` or `dblclick`, this happens to be OK because both handlers expect the same argument and are intended to be called on the same event i.e. a mesh is clicked/double-clicked; this is why we can just call all the subscribers for any given event type.